sled-agent/src/sim/sled_agent.rs

```diff
@@ -750,6 +752,17 @@ impl SledAgent {
                 self.config.hardware.reservoir_ram,
             )
             .context("reservoir_size")?,
+            disks: self
+                .storage
+                .lock()
```

Comment:
Are there any situations where this lock might be held for a long time (thinking tens of seconds or minutes)?

Reply:
I don't believe so, especially not since this is the simulated sled agent. At a cursory glance, i think it's mostly just getting locked to populate / grab contents from a group of hashmaps.